drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -436,7 +446,15 @@ def form_rules(**args)
     def form_rule(id:)
       raise 'not allowed' unless current_user.can_configure_data_collection?
 
-      Hmis::Form::Instance.find_by(id: id)
+      instance = Hmis::Form::Instance.find_by(id: id)
+      ensure_form_role_permission(instance.definition.role)
+
+      instance
+    end
+
+    field :non_admin_form_role, Forms::Enums::NonAdminFormRole, null: false
```

Comment:
I'd like to get rid of this. It's here because unless the type `Forms::Enums::NonAdminFormRole` is used somewhere in the api, it won't be part of the code gen process that makes it available in `HmisEnums` in the frontend, to be used in the dropdown when creating/editing a form. todo - is there a better way to convince the graphql codegen step that this should be part of the api?

Reply:
Unless I am misunderstanding with my comment https://github.com/greenriver/hmis-warehouse/pull/4531#discussion_r1675962139 I think we need to replace this with a backend picklist anyway. GR staff should be allowed to create forms for all roles